My Site Dashboard - Phase 1 - Remove `MySiteFragment`

ashiagr opened the pull request:
Fixes #15252

This PR removes the deprecated `MySiteFragment`, and also the feature flag `MySiteImprovementsFeatureConfig` used to guard the `ImprovedMySiteFragment` as we now always show `ImprovedMySiteFragment`.

Commits can be grouped as follows:
- Replace `MySiteFragment` with `ImprovedMySiteFragment` on bottom nav - ebf342b
- Remove references to`MySiteFragment` in `WPMainActivity` - e43f3e1 , 64a195c, f0772fc, 0cf03ec, 671a40c
- Delete `MySiteFragment` with its layout - 996d9cb, 669f8cf, 009c362, 64eaeeb
- Remove `MySiteImprovementsFeatureConfig.isEnabled()` guard - d5562ff, d407855, 5d3ee23, 4a0c025, 1c30cf2, 1418577, cf4ae26
- Remove duplicate tracking on complete quick start task/ tasks - 1418577 (See https://github.com/wordpress-mobile/WordPress-Android/issues/15250#issuecomment-907189777)

To test:
Smoke test "My Site" screen which now points to `ImprovedMySiteFragment`.

Also,
1. Notice that `App Settings` -> `Debug settings` -> `my_site_improvements_feature_flag` is not present anymore.
2. Toggle `QuickStartDynamicCardsFeatureConfig` for a site with "Quick Start" in progress and restart the app. Notice that the quick start section is appropriately shown based on the toggle state (with and without dynamic cards).

## Regression Notes
1. Potential unintended areas of impact
`MySiteFragment` was referenced from `WPMainActivity`, `Publicise`, `Reader` for showing quick start snackbars. It might be a good idea to test the quick start navigation with `ImprovedMySiteFragment` but there's a separate task to check `Quick Start - Guided Navigation` and we can add our findings to that task.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually

3. What automated tests I added (or what prevented me from doing so)
None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ashiagr committed:
Show ImprovedMySiteFragment on bottom nav

ashiagr committed:
Remove MySiteFragment reference on Fab click

shouldShowPublishPostQuickStartTask depends on getMySiteFragment, this logic can be removed. VM's onFabClicked takes care of corresponding actions for improved my site fragment.

ashiagr committed:
Remove logic in onPageChange for MySiteFragment

Only removed the else block meant for MySiteFragment. If block meant for ImprovedMySiteFragment is retained.

ashiagr committed:
Remove logic in onActivityResult for MySiteFragment

Only removed the if block meant for MySiteFragment. Else block meant for ImprovedMySiteFragment is retained.

ashiagr committed:
Remove showQuickStartDialog + positive, negative, neutral and touch outside implementation for MySiteFragment

Quick start dialog is now shown using showQuickStartDialog in ImprovedMySiteFragment.

ashiagr committed:
Remove now unused getMySiteFragment implementation

ashiagr committed:
Move constants referenced from MySiteFragment to MySiteViewModel

ashiagr committed:
Remove MySiteFragment injection from AppComponent

ashiagr committed:
Delete MySiteFragment :tada:

ashiagr committed:
Remove MySiteImprovementsFeatureConfig.isEnabled() guard in WPMainActivity

We now always show ImprovedMySiteFragment so this guard is not required. Also removed logic in mViewModel.getCreateAction() -> CREATE_NEW_POST invoked when MySiteImprovementsFeatureConfig is disabled as the config will now never be disabled.

ashiagr committed:
Remove MySiteImprovementsFeatureConfig.isEnabled() guard in Publicise list fragment

We now always show ImprovedMySiteFragment so this guard is not required.

ashiagr committed:
Remove MySiteImprovementsFeatureConfig.isEnabled() guard in Reader post list fragment

We now always show ImprovedMySiteFragment so this guard is not required.

ashiagr committed:
Remove MySiteImprovementsFeatureConfig.isEnabled() guard in PagesListViewModel

We now always show ImprovedMySiteFragment so this guard is not required.

ashiagr committed:
Remove duplicate tracking for completed task/ tasks from completeTaskAndRemindNextOne

ashiagr committed:
Remove MySiteImprovementsFeatureConfig.isEnabled() guard in WPMainActivityViewModel

We now always show ImprovedMySiteFragment so this guard is not required.

ashiagr committed:
Remove now unused onDismissByOutsideTouch in WPMainActivity

ashiagr committed:
Delete MySiteFragment layout :tada:

ashiagr committed:
Remove reference to ids in my_site_fragment.xml

ashiagr committed:
Remove now unused isTargetingBottomNavBar

ashiagr committed:
Remove improved my site fragment feature config from tracking

ashiagr committed:
Remove improved my site fragment feature config in build.gradle :tada:

ashiagr committed:
Analysis: Detekt - Remove empty default constructor

ashiagr committed:
Analysis: Detekt - Resolve multi line if else formatting

ashiagr committed:
Analysis: Lint - Remove unused resource

ashiagr committed:
Analysis: Lint - Remove unused dimen resources + styles

ashiagr committed:
Analysis: Lint - Remove unused string resource

ashiagr committed:
Tests: Fixed tests after MySiteImprovementsFeatureConfig is removed

Active quick start task should be PUBLISH_POST so that it can be completed on `MainActionListItem.ActionType.CREATE_NEW_POST` action (via create new page/post/story bottom sheet).

See WPMainActivityViewModel.onFabClicked:
```
_showQuickStarInBottomSheet.postValue(quickStartRepository.activeTask.value == PUBLISH_POST)
```

and

WPMainActivityViewModel.onCreateActionClicked

```
_showQuickStarInBottomSheet.value?.let { showQuickStart ->
    if (showQuickStart) {
       if (actionType == CREATE_NEW_POST) quickStartRepository.completeTask(PUBLISH_POST)
       _showQuickStarInBottomSheet.postValue(false)
    }
}
```

This is why added activeTask.value = PUBLISH_POST before calling onFabClicked.

Also, below test mentions "completes task" but verifies that `quickStartRepository.completeTask` is never invoked. Fixed the test to verify that `quickStartRepository.completeTask` is actually invoked.
```
CREATE_NEW_POST action in bottom sheet with active Quick Start completes task and hides the focus point
```

Earlier the tests passed as we always passed `shouldShowQuickStartFocusPoint` = true in tests. This variable was set from the activity when deprecated MySiteFragment existed, but is not  available any more and tests are adjusted to work with `ImprovedMySiteFragment`.

zwarm commented on WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java:
```diff
@@ -1158,38 +1154,12 @@ public void onClick(View v) {
-                        QuickStartUtils.cancelQuickStartReminder(this);
-                        AppPrefs.setQuickStartNoticeRequired(false);
-                        AppPrefs.setLastSkippedQuickStartTask(null);
-                        mPrivateAtomicCookie.clearCookie();
```
@ashiagr - Don't we still need to set `mPrivateAtomicCookie.clearCookie()` in this instance of a site change? (Maybe it's somewhere that I haven't seen yet - since I am reviewing commit by commit)

ashiagr replied:
Nice catch @zwarm! I've brought back the entire block in 15c664b. Seems like an existing bug where the code was not included for `ImprovedMySiteFragment`. 

I'm merging this PR, feel free to reopen in case you find anything odd. 

Thanks so much for the review! 🙇‍♀️

zwarm commented on WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt:
```diff
@@ -166,13 +161,7 @@ class WPMainActivityViewModel @Inject constructor(
-                        _completeBottomSheetQuickStartTask.call()
-                    }
-                }
+                if (actionType == CREATE_NEW_POST) quickStartRepository.completeTask(PUBLISH_POST)
```
@ashiagr - now I see what you were talking about in the tests. Nice. :)

zwarm commented on WordPress/src/main/java/org/wordpress/android/util/QuickStartUtilsWrapper.kt:
```diff
@@ -93,11 +93,9 @@ class QuickStartUtilsWrapper
         }
 
         quickStartStore.setDoneTask(siteLocalId, task, true)
-        analyticsTrackerWrapper.track(QuickStartUtils.getTaskCompletedTracker(task), mySiteImprovementsFeatureConfig)
```
Nice find - this is all done in QuickStartRepository now 👍

zwarm reviewed (approved):
🙇  @ashiagr, this was a tricky one with lots of little changes. It was very easy to review using the commits.

I am will approve, but **won't** merge because of that one question on clearing the atomic cookie. If it's safe to remove the clear cookie, then go ahead and merge. If not, then it needs to be added back.

I tested all button taps and they link to the correct pages. I have gone through the quick start from beginning to end, no surprises, guided navigation looks good too. I flipped the dynamic card feature flag on and tested and MySite still works correctly. I backgrounded the app and returned, that still works as expected. So far, all is good.

Thanks for the attention to detail 🚀

zwarm committed:
Remove reference to old my site fragment resource id

ashiagr committed:
Bring back WPMainActivity.onActivityResult -> SITE_PICKER code removed in commit f0772fca

This looks like an existing bug, where the code was included only for `getMySiteFragment()` (deprecated one) but should have been included for `ImprovedMySiteFragment` as well.

ashiagr committed:
Merge pull request #15266 from wordpress-mobile/issue/15252-remove-my-site-fragment-connected-tests

My Site Dashboard - Phase 1: Address connected tests issues
